docs/linux/setup_linux-host_qemu-vm_riscv64-kernel.md

```diff
@@ -0,0 +1,153 @@
+# Setup: Debian/Ubuntu host, QEMU vm, riscv64 kernel
+
+# GCC
+
+Obtain `riscv64-linux-gnu-gcc` and `riscv-unknown-elf-gcc`, at least GCC version 8. The latest
+Debian/Ubuntu distributions should provide both cross-compilers in a recent enough version.
+Alternatively, you can also build your own
+[RISC-V GNU compiler toolchain](https://github.com/riscv/riscv-gnu-toolchain) from source
+
+# Kernel
+
+The following instructions were tested with Linux Kernel `v5.8-rc2`. In addition you need the
+["riscv: Allow building with kcov coverage"](https://lore.kernel.org/linux-riscv/20200626124056.29708-1-tklauser@distanz.ch/)
+patch. Create a kernel config with:
+
+```shell
+make ARCH=riscv CROSS_COMPILE=riscv64-linux-gnu- defconfig
+```
+
+In addition enable the [recommended Kconfig options for syzkaller](/docs/linux/kernel_configs.md).
+
+Then build kernel with:
+
+```
+make ARCH=riscv CROSS_COMPILE=riscv64-linux-gnu- -j $(nproc) vmlinux
+```
+
+# BBL (Berkeley Boot Loader)
+
+Clone the BBL repository and build the bootable image containg bootloader and kernel:
+
+```shell
+git clone https://github.com/riscv/riscv-pk
+cd riscv-pk
+mkdir build && cd build
+../configure --enable-logo --host=riscv64-unknown-elf --with-payload=/path/to/linux/vmlinux
```

Comment:
Since this compiler is broken in my distro, I tried `--host=riscv64-linux-gnu` and at least the build worked. Do we really need the second compiler? Building buildroot, so I don't know yet if it will work or not. But maybe you know the answer already?

Reply:
Yes, the `gcc-riscv64-unknown-elf` compiler is needed to build the boot loader. This is to compiler bare-metal binaries (like bootloaders) as it e.g. does not link against the libc.

I'll try to figure out a way to provide an easier solution to this.

Reply:
Sorry, nevermind my previous comment.

The README in https://github.com/riscv/riscv-pk even mentions that `--host=riscv64-linux-gnu` could be used. Just tested it on my setup and it seems to work. Will adjust the docs and drop the parts related to `riscv64-unknown-elf-gcc`.